contracts/mocks/StakingPool/SPMockStakingPoolLibrary.sol

```diff
@@ -0,0 +1,27 @@
+// SPDX-License-Identifier: GPL-3.0-only
+
+pragma solidity ^0.8.16;
+
+/**
+ * @dev Simple library to derive the staking pool address from the pool id without external calls
+ */
+contract SPMockStakingPoolLibrary {
+
+  function getAddress(address factory, uint poolId) public pure returns (address) {
+
+    bytes32 hash = keccak256(
+      abi.encodePacked(
+        hex'ff',
+        factory,
+        poolId, // salt
+      // init code hash of the MinimalBeaconProxy
+      // updated using patch-staking-pool-library.js script
+        hex'203b477dc328f1ceb7187b20e5b1b0f0bc871114ada7e9020c9ac112bbfb6920' // init code hash
+      )
+    );
+
+    // cast last 20 bytes of hash to address
+    return address(uint160(uint(hash)));
```

Comment:
Import and use the library instead. Also this would better be the `Cover` mock:
1. Cover already has an external `stakingPool(uint)` function so it would make sense mocking it instead of adding a new contract (which technically doesn't exist).
2. Cover is the operator of the NFT contracts

Reply:
BTW, you've put it in the wrong folder. Should be `mocks/StakingNFT`?